### PR TITLE
Add demo link to the docs

### LINKED ISSUE #1224: Add Live demo button to the docs and GitHub repo
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

As discussed on the call yesterday, we should also add Live demo button on Toolpad docs and Toolpad GitHub repo.

Tracking these sources in GA would be helpful too.

Relates to: https://github.com/mui/mui-toolpad/issues/1041

### Examples 🌈

_No response_

### Motivation 🔦

_No response_

### PR DESCRIPTION
**Pull request by bharatkashyap**
- Closes #1224 
- Visual demonstration 
Screenshot 2022-11-09 at 9 00 52 PM

https://deploy-preview-1319--mui-toolpad-docs.netlify.app/toolpad/getting-started/overview/

### CODE REVIEW

File: docs/src/content/upvote.js [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-const upvote = {
-  title: '👍 Upvote the features you want to get prioritized',
-  description:
-    'Want to request a feature? Head over to our repo and if not already listed, feel free to open an issue explaining the use case.',
-  action: {
-    href: ROUTES.toolpadUpvote,
-    label: 'Upvote',
-  },
-};
+export const title = '👍 Upvote the features you want to get prioritized';
```
Comment (apedroferreira) [inline docs/src/content/upvote.js:3]: Did you change this pattern to enable tree shaking?
Eventually we could have other files follow the same pattern then for things to be consistent, but it's probably not a high priority.

File: docs/data/toolpad/getting-started/overview.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 <p class="description">Toolpad is a low-code UI builder for creating apps with MUI's suite of component libraries.</p>
 
+{{"component": "components/landing/Banner.js", "action": "tryDemo", "category": "ToolpadDocs", "description": "Check out Toolpad's interactive demo. No sign-up required!", "href": "https://demo.toolpad.io", "label": "Try Demo", "docs": true }}
```
Comment (apedroferreira) [inline docs/data/toolpad/getting-started/overview.md:5]: How does this work? :D Is it intended?

**Review by apedroferreira — APPROVED** [review]
Looks good!
I can't test any preview but form the screenshots looks great, as long as responsiveness works well too.

File: docs/src/content/upvote.js (hunk 1 above)
Comment (bharatkashyap) [reply docs/src/content/upvote.js:3]: I did this to allow the `<Banner>` component to be used in the docs as well, using the `{{ component: ...}}` syntax, where I wasn't aware of a method to pass an object as props here https://github.com/mui/mui-toolpad/pull/1319#discussion_r1021933417

File: docs/data/toolpad/getting-started/overview.md (hunk 2 above)
Comment (bharatkashyap) [reply docs/data/toolpad/getting-started/overview.md:5]: This happens through https://github.com/mui/material-ui/blob/master/docs/packages/markdown/loader.js where we allow `{{ "component": .. }}` and `{ "demo": ...}}` to support components inside the docs

File: docs/src/content/upvote.js (hunk 1 above)
Comment (apedroferreira) [reply docs/src/content/upvote.js:3]: Ah, sounds good then!

**Comment by apedroferreira** [comment]
> Just added the [deploy preview link](https://deploy-preview-1319--mui-toolpad-docs.netlify.app/toolpad/getting-started/overview/) to the PR description as well

Thanks!
This screen width looks kinda weird, maybe not using horizontal margins in small screens will fix it?

Screenshot 2022-11-15 at 16 24 14

**Review by apedroferreira — CHANGES_REQUESTED** [review]
Marking as "requested changes" just to track the responsiveness issue.

**Comment by bharatkashyap** [comment]
@apedroferreira Added some changes to the horizontal padding (as you correctly mentioned) which seems to result in an appropriate banner on the docs pages across screen widths:


https://user-images.githubusercontent.com/19550456/202415108-0264dfe8-f4dc-4d5d-a986-f45dd1495284.mov

**Review by apedroferreira — APPROVED** [review]
Looks good now, thanks for the changes!

File: docs/data/toolpad/getting-started/overview.md (hunk 2 above)
Comment (oliviertassinari) [inline docs/data/toolpad/getting-started/overview.md:5]: I'm not sure that we should use marketing pages components `{{"component": "components/landing/Banner.js"` inside the docs page. It looks like this now:

Screenshot 2022-12-09 at 13 07 08

Maybe a banner would feel better like this for the docs:

Screenshot 2022-12-09 at 13 07 43

**Review by oliviertassinari — COMMENTED** [review]
great idea to make the demo easier to access 👍 , to keep the visitor hooked before they leave.

**Pull request merged**